Mathlib/CategoryTheory/Category/Twop.lean

```diff
@@ -0,0 +1,220 @@
+  toTwoPointing : TwoPointing X
+set_option linter.uppercaseLean3 false in
+#align Twop Twop
+
+namespace Twop
+
+instance : CoeSort Twop (Type _) :=
+  ⟨X⟩
+
+-- attribute [protected] Twop.X -- Porting note: protected attribute doesn't work
+
+/-- Turns a two-pointing into a two-pointed type. -/
+def of {X : Type _} (to_two_pointing : TwoPointing X) : Twop :=
+  ⟨X, to_two_pointing⟩
+set_option linter.uppercaseLean3 false in
+#align Twop.of Twop.of
+
+@[simp]
+theorem coe_of {X : Type _} (to_two_pointing : TwoPointing X) : ↥(of to_two_pointing) = X :=
+  rfl
+set_option linter.uppercaseLean3 false in
+#align Twop.coe_of Twop.coe_of
+
+alias of ← _root_.TwoPointing.Twop
+set_option linter.uppercaseLean3 false in
+#align two_pointing.Twop TwoPointing.Twop
+
+instance : Inhabited Twop :=
+  ⟨of TwoPointing.bool⟩
+
+/-- Turns a two-pointed type into a bipointed type, by forgetting that the pointed elements are
+distinct. -/
+noncomputable def toBipointed (X : Twop) : Bipointed :=
+  X.toTwoPointing.toProd.Bipointed
+set_option linter.uppercaseLean3 false in
+#align Twop.to_Bipointed Twop.toBipointed
+
+@[simp]
+theorem coe_toBipointed (X : Twop) : ↥X.toBipointed = ↥X :=
+  rfl
+set_option linter.uppercaseLean3 false in
+#align Twop.coe_to_Bipointed Twop.coe_toBipointed
+
+noncomputable instance largeCategory : LargeCategory Twop :=
+  InducedCategory.category toBipointed
+set_option linter.uppercaseLean3 false in
+#align Twop.large_category Twop.largeCategory
+
+noncomputable instance concreteCategory : ConcreteCategory Twop :=
+  InducedCategory.concreteCategory toBipointed
+set_option linter.uppercaseLean3 false in
+#align Twop.concrete_category Twop.concreteCategory
+
+noncomputable instance hasForgetToBipointed : HasForget₂ Twop Bipointed :=
+  InducedCategory.hasForget₂ toBipointed
+set_option linter.uppercaseLean3 false in
+#align Twop.has_forget_to_Bipointed Twop.hasForgetToBipointed
+
+
+/-- Swaps the pointed elements of a two-pointed type. `TwoPointing.swap` as a functor. -/
+@[simps]
+noncomputable def swap : Twop ⥤ Twop where
+  obj X := ⟨X, X.toTwoPointing.swap⟩
+  map f := ⟨f.toFun, f.map_snd, f.map_fst⟩
+set_option linter.uppercaseLean3 false in
+#align Twop.swap Twop.swap
+
+/-- The equivalence between `Twop` and itself induced by `Prod.swap` both ways. -/
+@[simps!]
+noncomputable def swapEquiv : Twop ≌ Twop :=
+  CategoryTheory.Equivalence.mk swap swap
+    (NatIso.ofComponents
+      (fun X =>
+        { hom := ⟨id, rfl, rfl⟩
+          inv := ⟨id, rfl, rfl⟩ })
+      fun f => rfl)
+    (NatIso.ofComponents
+      (fun X =>
+        { hom := ⟨id, rfl, rfl⟩
+          inv := ⟨id, rfl, rfl⟩ })
+      fun f => rfl)
+set_option linter.uppercaseLean3 false in
+#align Twop.swap_equiv Twop.swapEquiv
+
+@[simp]
+theorem swapEquiv_symm : swapEquiv.symm = swapEquiv :=
+  rfl
+set_option linter.uppercaseLean3 false in
+#align Twop.swap_equiv_symm Twop.swapEquiv_symm
+
+end Twop
+
+@[simp]
+theorem twop_swap_comp_forget_to_bipointed :
```

Comment:
```suggestion
theorem twop_swap_comp_forget_to_Bipointed :
```
I'm actually not at all sure about this one, and would like to know what is more consistent with what we've been doing so far.